### DISCUSSION TITLE
SyntheticTimeoutException: what is this, how to suppress or ignore them?

### DISCUSSION THREAD

**Discussion by digital88** [ZiggyCreatures/FusionCache#307] [Q&A] [answered]
Hi.

I have no idea how to handle SyntheticTimeoutException, is there any way to ignore or suppress them? Also, whats the root cause of said exception, and what happens if I ignore or suppress them?

**Comment by jodydonetti** [comment, marked as answer]
Hi @digital88 , did you see them in the logs or are you getting them thrown up to your own code?

Anyway `SyntheticTimeoutException`s are thrown when you enable **hard** [timeouts](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/Timeouts.md) and those timeouts actually happen.

Have you enabled the feature maybe?
If so, basically you are saying _"during a refresh, if the factory takes more than X time, I prefer to receive an exception instead of waiting too much"_ because you prefer to be fast and handle the error yourself.

If you enabled **soft** timeout instead, you will not receive them in your code and just see some warnings in the logs. Soft timeout means _"during a refresh, if the factory takes more than X time, I prefer to temporarily reuse a stale (expired) value instead of waiting too much"_.

Warning levels are also configurable (via `FusionCacheOptions`), so you can "suppress" them by setting the related log level to trace or similar.

Hope this helps, let me know.

**Comment by digital88** [comment]
> did you see them in the logs or are you getting them thrown up to your own code?

Sentry catches them and reports, so I assume they bubble up to my code. They rarely do tho.

> Have you enabled the feature maybe?

Yes, as suggested in [StepByStep](https://github.com/ZiggyCreatures/FusionCache/blob/main/docs/StepByStep.md) guide, I did enable them, and also enabled soft timeouts. The guide actually mentions `SyntheticTimeoutException` but I had no idea what to do to handle them, or if I need to do anything at all. The timeouts doc you linked explains it.

So I guess I remove hard timeout from config entirely, I'm ok with getting stale (expired) cache value.

Thank you for answering, you really helped me to better understand soft/hard timeouts.

**Reply by jodydonetti** [reply]
Happy to have helped, and you helped me too: now that you mentioned it, I think I'll add a paragraph in the Timeouts docs to explicitly state what happens when enabling hard timeouts, including the exception thrown.
Thanks!
